socorro/cron/jobs/featured_versions_automatic.py

```diff
@@ -0,0 +1,138 @@
+class FeaturedVersionsAutomaticCronApp(BaseCronApp):
+    app_name = 'featured-versions-automatic'
+    app_description = """Use https://product-details.mozilla.org/1.0/
+    to automatically figure out what product versions should be
+    set as featured.
+    """
+
+    required_config = Namespace()
+    required_config.add_option(
+        'api_endpoint_url',
+        default=(
+            'https://product-details.mozilla.org/1.0/'
+        ),
+        doc='URL from which we can download all the featured versions'
+    )
+    required_config.add_option(
+        'products',
+        default='firefox,mobile,thunderbird',
+        from_string_converter=lambda line: tuple(
+            [x.strip() for x in line.split(',') if x.strip()]
+        ),
+        doc='a comma-delimited list of products to recognize'
+    )
+    required_config.add_option(
+        'aliases',
+        default='mobile:FennecAndroid',
+        from_string_converter=alias_list_to_dict,
+        doc=(
+            'a comma-delimited list of name:RealName. If a product '
+            'does not need an alias match it by capitalizing the name '
+            '(e.g. "thunderbird" -> "Thunderbird")'
+        )
+    )
+
+    def run(self, connection):
+        assert isinstance(self.config.api_endpoint_url, basestring)
+        assert self.config.api_endpoint_url.endswith('/')
+
+        # The @with_single_postgres_transaction decorator makes
+        # sure this cursor is committed or rolled back and cleaned up.
+        cursor = connection.cursor()
+
+        for product in self.config.products:
+            url = '{}{}_versions.json'.format(
+                self.config.api_endpoint_url,
+                product
+            )
+            response = requests.get(url)
+            if response.status_code != 200:
+                raise DownloadError(
+                    '{} ({})'.format(
+                        url,
+                        response.status_code,
+                    )
+                )
+            versions = response.json()
+            self._set_featured_versions(
+                cursor,
+                product,
+                versions,
+            )
+
+    def _set_featured_versions(self, cursor, product, versions):
+        # 'product_name' is what's it's called in our product_versions
+        # table.
+        product_name = self.config.aliases.get(
+            product,
+            product.capitalize()
+        )
+        featured = set()
+        if product_name == 'Firefox':
+            featured.add(versions['FIREFOX_NIGHTLY'])
+            featured.add(versions['FIREFOX_AURORA'])
+            featured.add(versions['LATEST_FIREFOX_VERSION'])
+            beta = versions['LATEST_FIREFOX_DEVEL_VERSION']
+            # If the beta version is something like '59.0b12'
+            # then convert that to '59.0b' which is basically right-stripping
+            # any numbers after the 'b'.
+            beta = re.sub('b(\d+)$', 'b', beta)
```

Comment:
The regex makes sure the `b\d+` really is the end of the string. Can't do that with the split. No biggie.